WordPress/Classes/Models/AbstractPost+Local.swift

```diff
@@ -9,4 +9,14 @@ extension AbstractPost {
     var isLocalRevision: Bool {
         return self.originalIsDraft() && self.isRevision() && self.remoteStatus == .local
     }
+
+    /// Count posts that have never been uploaded to the server.
+    ///
+    /// - Parameter context: A `NSManagedObjectContext` in which to count the posts
+    /// - Returns: number of local posts in the given context.
+    static func countLocalPosts(using context: NSManagedObjectContext) -> Int {
+        let request = NSFetchRequest<AbstractPost>(entityName: NSStringFromClass(AbstractPost.self))
+        request.predicate = NSPredicate(format: "postID = NULL OR postID <= 0")
+        return (try? context.count(for: request)) ?? 0
+    }
```

Comment:
Have you considered adding a test for this method? 

I'm particularly wary of the hardcoded `postID` column/property. Even though it's unlikely, I'm not sure what would happen here if we changed the name. Presumably the code would crash?

Reply:
I can certainly add a couple of simple unit tests. 👍 

It's probably going to crash if the property name is changed. I assumed that changing property name is not something we would do, since I've seen lots of places that uses hardcoded property names in the predicate strings.

Reply:
I want to use a function in my other changes. I'm going to merge this PR and make a follow up PR to add tests.

Reply:
Sure :)

Reply:
A simple unit test is added [here](https://github.com/wordpress-mobile/WordPress-iOS/pull/19354/commits/9ac528a9d7b63c91ad4e46d20a9ee4160e08d403).